2-mandatory/8-password-validator.js

```diff
@@ -23,7 +23,17 @@ PasswordValidationResult=  [false, false, false, false, true]
 
 */
 
-function validatePasswords(passwords) {}
+function validatePasswords(passwords) {
+  return passwords.map(
+    (x, y) =>
+      x.length > 4 &&
+      containsUppercaseLetter(x) &&
+      containsLowercaseLetter(x) &&
+      containsNumber(x) &&
+      containsSymbol(x) &&
+      y === passwords.indexOf(x)
```

Comment:
This is a nice simple solution. I would try to give the vars more descriptive names though as it can be hard to figure out what's going on here if you don't happen to rember the syntax of the map function off the top of your head